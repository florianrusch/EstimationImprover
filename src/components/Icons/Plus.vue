<template>
	<svg
		width="100%"
		height="100%"
		viewBox="0 0 96 96"
		version="1.1"
		xmlns="http://www.w3.org/2000/svg"
		xmlns:xlink="http://www.w3.org/1999/xlink"
		xml:space="preserve"
		xmlns:serif="http://www.serif.com/"
		style="
			fill-rule: evenodd;
			clip-rule: evenodd;
			stroke-linecap: round;
			stroke-linejoin: round;
			stroke-miterlimit: 1.5;
		"
	>
		<title>{{ title }}</title>
		<desc v-if="description">{{ description }}</desc>
		<g>
			<path d="M48,7.5l0,81" style="fill: none; stroke: #000; stroke-width: 10px" />
			<path d="M7.5,48l81,0" style="fill: none; stroke: #000; stroke-width: 10px" />
		</g>
	</svg>
</template>

<script lang="ts">
import { defineComponent } from 'vue'

export default defineComponent({
	setup() {
		return {
			title: 'Plus-Icon',
			description: 'This is an icon which show an plus symbol.',
		}
	},
})
</script>

<style scoped>
svg {
	fill: currentColor;
	line-height: 1;
	height: 1em;
	width: 1em;
}
</style>
