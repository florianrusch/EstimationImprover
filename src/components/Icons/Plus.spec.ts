import { mount } from '@vue/test-utils'
import Plus from './Plus.vue'

describe('# Plus', () => {
	it('renders correctly', () => {
		// given
		const wrapper = mount(Plus)

		// when
		// then
		expect(wrapper.element).toMatchSnapshot()
	})
})
