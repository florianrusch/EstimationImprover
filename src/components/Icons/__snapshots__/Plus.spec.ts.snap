// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`# Plus renders correctly 1`] = `
<svg
  height="100%"
  style="fill-rule: evenodd; clip-rule: evenodd; stroke-linecap: round; stroke-linejoin: round; stroke-miterlimit: 1.5;"
  version="1.1"
  viewBox="0 0 100 100"
  width="100%"
  xml:space="preserve"
  xmlns="http://www.w3.org/2000/svg"
  xmlns:serif="http://www.serif.com/"
  xmlns:xlink="http://www.w3.org/1999/xlink"
>
  <title>
    Plus-Icon
  </title>
  <desc>
    This is an icon which shows a plus symbol.
  </desc>
  <g>
    <path
      d="M 50,0 l 0,100"
      style="fill: none; stroke: currentColor; stroke-width: 10px;"
    />
    <path
      d="M 0,50 l 100,0"
      style="fill: none; stroke: currentColor; stroke-width: 10px;"
    />
  </g>
</svg>
`;
